src/app/forum/containers/thread/thread.component.ts

```diff
@@ -157,32 +169,76 @@ export class ThreadComponent implements AfterViewInit, OnDestroy {
     );
 
     this.subscriptions.add(
-      this.isThreadStatusOpened$.pipe(delay(0)).subscribe(isThreadStatusOpened => {
-        if (!isThreadStatusOpened) {
-          this.form.get('messageToSend')?.disable();
+      combineLatest([
+        this.threadStatus$.pipe(delay(0)),
+        this.disableControls$,
+      ]).subscribe(([ status, disableControls ]) => {
+        if (status?.data && (status.data.open || status.data.canOpen) && !disableControls) {
+          this.form.get('messageToSend')?.enable();
           return;
         }
-        this.form.get('messageToSend')?.enable();
-        this.messageToSendEl?.nativeElement.focus();
+        this.form.get('messageToSend')?.disable();
       })
     );
+    this.subscriptions.add(
+      this.isThreadStatusOpened$.pipe(delay(0), filter(isThreadStatusOpened => isThreadStatusOpened)).subscribe(() =>
+        this.messageToSendEl?.nativeElement.focus()
+      )
+    );
   }
 
   ngOnDestroy(): void {
     this.subscriptions.unsubscribe();
   }
 
-  sendMessage(threadId: ThreadId): void {
+  clearMessageToSendControl(): void {
+    this.form.reset({
+      messageToSend: '',
+    });
+  }
+
+  sendMessage(threadId: ThreadId, isThreadOpened: boolean): void {
     const messageToSend = this.form.value.messageToSend?.trim();
     if (!messageToSend) return;
-    this.store.select(forum.selectThreadToken).pipe(
+    this.disableControls$.next(true);
+
+    const threadToken$ = this.store.select(forum.selectThreadToken).pipe(
       take(1), // send on the current thread (if any) only
       filter(isNotUndefined),
-    ).subscribe(token => this.forumWebsocketClient.send(publishEventsAction(token, [ messageEvent(messageToSend) ])));
-    this.updateThreadService.update(threadId.itemId, threadId.participantId, { messageCountIncrement: 1 }).subscribe();
-    this.form.reset({
-      messageToSend: '',
+    );
+
+    if (isThreadOpened) {
+      threadToken$.subscribe({
+        next: token => this.forumWebsocketClient.send(publishEventsAction(token, [ messageEvent(messageToSend) ])),
+        error: () => this.disableControls$.next(false),
+      });
+      this.updateThreadService.update(threadId.itemId, threadId.participantId, { messageCountIncrement: 1 }).subscribe();
+      this.clearMessageToSendControl();
+      this.disableControls$.next(false);
+      return;
+    }
+
+    this.changeThreadStatus({
+      open: true,
+      threadId,
+      messageCountIncrement: 1,
+    }).subscribe({
+      error: () => this.disableControls$.next(false),
     });
+    this.subscriptions.add(
```

Comment:
again here, you are not waiting for the thread to be opened to do this part.

Reply:
`forum.threadPanelActions.threadStatusChanged()` - calls `threadStatusChanged` action, then I catch it in next line: `forum.selectThreadStatusOpen`;

This is the way of ngrx, we could not have `changeThreadStatus` method as observable (but just call the action), and so we had to wait somehow status - as I done here.

I can put `forum.selectThreadStatusOpen` under switch map of `changeThreadStatus`, but I don't see the reason as the prev status anyway will be false by condition above and so `selectThreadStatusOpen` will be true after status has changed.

A bit implicitl but this is store way

Reply:
Yes sure, I was probably tired when reading it last time.